File: Day_1/Dhruv/lc11.py
#leetcode problem 11. Container With Most Water 

class Solution(object):
    def maxArea(self, height):
        low = 0
        high = len(height) - 1
        max_area = 0

        while low < high:
            length = min(height[low], height[high])
            width = high - low
            max_area = max(max_area, length * width)

            if height[low] < height[high]:
                low+=1
            else:
                high-=1
        
        return max_area

File: Day_15/Dhruv/lc11.java
//leetcode 11. Container With Most Water

class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right= height.length-1;
        int max_area=0;
        while(left<right){
            int length=Math.min(height[left],height[right]);
            int width=right-left;
            max_area= Math.max(max_area, length * width);
            
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return max_area;
    }
}